src/components/category.vue

```diff
@@ -79,8 +94,27 @@ export default {
     },
     hasResults() {
       return this.emojis.length > 0
+    },
+    emojiObjects() {
+      return this.emojis.map((emoji) => {
+          let emojiObject = new EmojiData(
+            emoji, this.emojiProps.skin, this.emojiProps.set, this.data)
```

Comment:
Emoji data is now wrapped into the `EmojiData` object to unify access and reduce calls to `getData`